File: c++/15828.cpp
#include <iostream>
#include <queue>

using namespace std;

queue<int> q;

int main() {
  freopen("example.txt", "r", stdin);

  int N;
  cin >> N;

  while (true) {
    int a;
    cin >> a;

    if (a > 0) {
      if (q.size() < N) q.push(a);
    } else if (a == 0) {
      q.pop();
    } else if (a == -1) {
      if (q.size() == 0)
        cout << "empty";
      else {
        while (!q.empty()) {
          cout << q.front() << ' ';
          q.pop();
        }
      }
      break;
    }
  }

  return 0;
}

File: js/15828.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map(Number);

const N = inputs[0];

let idx = 0;
const q = [];

for (let i = 1; i <= inputs.length; i++) {
  if (inputs[i] > 0) {
    if (q.length < N + idx) q.push(inputs[i]);
  } else if (inputs[i] === 0) {
    idx++;
  } else if (inputs[i] === -1) {
    if (q.length === idx) console.log('empty');
    else {
      const ans = [];
      while (q.length !== idx) {
        ans.push(q[idx++]);
      }
      console.log(ans.join(' '));
    }
  }
}
